query-engine/driver-adapters/src/conversion/mysql.rs

```diff
@@ -0,0 +1,42 @@
+use crate::conversion::JSArg;
+use serde_json::value::Value as JsonValue;
+
+const DATE_FORMAT: &str = "%Y-%m-%d %H:%M:%S";
+const TIME_FORMAT: &str = "%H:%M:%S";
+
+pub fn value_to_js_arg(value: &quaint::Value) -> serde_json::Result<JSArg> {
+    let res = match &value.typed {
+        quaint::ValueType::Json(s) => match s {
+            Some(ref s) => {
+                let json_str = serde_json::to_string(s)?;
+                JSArg::RawString(json_str)
+            }
+            None => JsonValue::Null.into(),
+        },
+        quaint::ValueType::Bytes(bytes) => match bytes {
+            Some(bytes) => JSArg::Buffer(bytes.to_vec()),
+            None => JsonValue::Null.into(),
+        },
+        quaint::ValueType::Numeric(bd) => match bd {
+            // converting decimal to string to preserve the precision
+            Some(bd) => JSArg::RawString(bd.to_string()),
+            None => JsonValue::Null.into(),
+        },
+        quaint::ValueType::DateTime(dt) => match dt {
+            Some(dt) => JSArg::RawString(dt.format(DATE_FORMAT).to_string()),
+            None => JsonValue::Null.into(),
+        },
+        quaint::ValueType::Time(dt) => match dt {
+            Some(dt) => JSArg::RawString(dt.format(TIME_FORMAT).to_string()),
+            None => JsonValue::Null.into(),
+        },
```

Comment:
Maybe out of scope of this PR since a lot of this code is moved around as is, and could be changed in a follow-up PR, but I'd get rid of the duplicate `None => JsonValue::Null.into()` match arms and inner `match` statements altogether by making `Some` a part of the outer pattern. The last catch-all pattern will handle `None`s for any `ValueType`.

```suggestion
        quaint::ValueType::Json(Some(s)) => {
            let json_str = serde_json::to_string(s)?;
            JSArg::RawString(json_str)
        }
        quaint::ValueType::Bytes(Some(bytes)) => JSArg::Buffer(bytes.to_vec()),
        quaint::ValueType::Numeric(Some(bd)) => {
            // converting decimal to string to preserve the precision
            JSArg::RawString(bd.to_string())
        }
        quaint::ValueType::DateTime(Some(dt)) => JSArg::RawString(dt.format(DATE_FORMAT).to_string()),
        quaint::ValueType::Time(Some(dt)) => JSArg::RawString(dt.format(TIME_FORMAT).to_string()),
```

Reply:
That could be definitely some future improvement.